### ISSUE TITLE
use swap-head to inference a dense model?

### ISSUE THREAD

**Issue by Sairampv98** [RobertCsordas/moe_attention#1]
I am curious about if its possible to inference a dense model using swap head attention mask (something like flashAttention) or is this used to train new models from scratch ?

**Comment by RobertCsordas** [comment]
No, using this without retraining in its current form is impossible. It is not just masking, but it swaps some attention matrices dynamically. Developing a fine-tuning method for this might be possible, but that is an open research question.
